lib/logstash/inputs/kafka.rb

```diff
@@ -36,10 +36,19 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   # the same consumer group.
   config :group_id, :validate => :string, :default => 'logstash'
   # The topic to consume messages from
-  config :topic_id, :validate => :string, :required => true
+  config :topic_id, :validate => :string, :default => nil
```

Comment:
This was somewhat why I initially made topic_id required and then made a blacklist toggle but I like the current way better.  It is more straightforward as to what you are configuring while the other kinda hid the fact that if you didn't have blacklist turned on, topic_id was treated as a whitelist and if it was on then topic_id was a blacklist.